tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SingleMessageTcpListener.cs

```diff
@@ -67,15 +62,27 @@ await AcceptTcpClientAsync(_tcpListener, cancellationToken)
         /// the cancellation token is registered to stop the listener.
         /// </summary>
         /// <remarks>See https://stackoverflow.com/questions/19220957/tcplistener-how-to-stop-listening-while-awaiting-accepttcpclientasync</remarks>
-        private static async Task<TcpClient> AcceptTcpClientAsync(TcpListener listener, CancellationToken token)
+        private static async Task<TcpClient> AcceptTcpClientAsync(TcpListener listener, CancellationToken ct)
         {
-            try
-            {
-                return await listener.AcceptTcpClientAsync().ConfigureAwait(false);
-            }
-            catch (Exception ex) when (token.IsCancellationRequested)
+            using (ct.Register(listener.Stop))
             {
-                throw new OperationCanceledException("Cancellation was requested while awaiting TCP client connection.", ex);
+                try
```

Comment:
I will move this logic into the product SingleMessageTcpListener.cs or consolidate the 2 classes.